storagesystem/util.go

```diff
@@ -239,13 +234,28 @@ func (r *rcloneSeeker) Seek(offset int64, whence int) (int64, error) {
 	case io.SeekEnd:
 		r.offset = r.size + offset
 	default:
-		return 0, errors.New("Unknown seek mode")
+		return 0, errors.New("unknown seek mode")
 	}
 	if r.offset > r.size {
-		return 0, errors.New("Seeking past end of file")
+		return 0, errors.New("seeking past end of file")
 	}
 	if r.offset < 0 {
-		return 0, errors.New("Seeking before start of file")
+		return 0, errors.New("seeking before start of file")
+	}
+	if r.file != nil {
+		var err error
+		if r.offset > prevOffset {
+			// Handle forward seek.
+			_, err = io.CopyN(io.Discard, r.file, r.offset-prevOffset)
```

Comment:
I think this might be a problem, because an `http.ServeContent` does a seek-end + seek-start to determine the size before working out what the http ranges mean. So the first use of this is going to discard the entire rclone reader. And for what purpose? Does it need to perform a full read to work?

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/net/http/fs.go;l=196-200 is `ServeContent`'s seeks btw

Reply:
see comment above, I don't think the seek-end is a problem, however I'm still not sure this copy discard is a good idea; at least I'm not sure what the purpose of it is

Reply:
The call to `http.ServeContent` is done before, and only before, and ranges are retrieved/read and simply moves `r.offset` and returns the size from the database. The section of code above is not affected.

Reply:
OK, I removed this so that the code does not continue reading from the same stream (pipe reader) following a `Seek`.